doc/guides/beyond-GSG.rst

```diff
@@ -0,0 +1,277 @@
+See `Installing Packages`_ in the Python Packaging User Guide for more
+information about pip\ [#pip]_, including `information on -\\-user`_.
+
+- On Linux, make sure ``~/.local/bin`` is at the front of your :envvar:`PATH`
+  :ref:`environment variable <env_vars>`, or programs installed with ``--user``
+  won't be found. Installing with ``--user`` avoids conflicts between pip
+  and the system package manager, and is the default on Debian-based
+  distributions.
+
+- On macOS, `Homebrew disables -\\-user`_.
+
+- On Windows, see the `Installing Packages`_ information on ``--user`` if you
+  require using this option.
+
+On all operating systems, pip's ``-U`` flag installs or updates the package if the
+package is already installed locally but a more recent version is available. It
+is good practice to use this flag if the latest version of a package is
+required.  (Check the :zephyr_file:`scripts/requirements.txt` file to
+see if a specific Python package version is expected.)
+
+Advanced Setup and tool chain alternatives
+******************************************
+
+Here are some alternative instructions for more advanced platform setup
+configurations for supported development platforms:
+
+.. toctree::
+   :maxdepth: 1
+
+   Linux setup alternatives <../getting_started/installation_linux.rst>
+   macOS setup alternatives <../getting_started/installation_mac.rst>
+   Windows setup alternatives <../getting_started/installation_win.rst>
+
+.. _gs_toolchain:
+
+Set Up a Toolchain
+******************
+
+Zephyr binaries are compiled and linked by a *toolchain* comprised of
+a cross-compiler and related tools which are different than the compiler
+and tools used for developing software that runs
+natively on your operating system.
+
+On Linux systems, you can install the :ref:`Zephyr SDK <zephyr_sdk>` to
+get toolchains for all supported architectures.  Otherwise, you can
+install other toolchains in the usual way for your operating system: with
+installer programs or system package managers, by downloading and
+extracting a zip archive, etc.
+
+You configure the Zephyr build system to use a specific toolchain by
+setting :ref:`environment variables <env_vars>` such as
+:envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to a supported value, along with
+additional variable(s) specific to the toolchain variant.
+
+While the Zephyr SDK includes standard tool chains for all supported
+architectures, there are also customized alternatives as described in
+these documents. (If you're not sure which to use, check your specific
+:ref:`board-level documentation <boards>`. If you're targeting an Arm
+Cortex-M board, for example, :ref:`toolchain_gnuarmemb` is a safe bet.)
+
+.. toctree::
+   :maxdepth: 2
+
+   ../getting_started/toolchain_3rd_party_x_compilers.rst
+   ../getting_started/toolchain_other_x_compilers.rst
+   ../getting_started/toolchain_host.rst
+   ../getting_started/toolchain_custom_cmake.rst
+
+Cloning the Zephyr Repositories
```

Comment:
I like the idea of giving more detail in this guide than in the GSG, but it must be made clearer what is repetition and what is additional content. Also, we should have links from each section in the GSG to the "more information if you need it" section here.